python/ray/exceptions.py

```diff
@@ -154,8 +154,8 @@ class RayActorError(RayError):
     """
 
     def __str__(self):
-        return ("The actor died unexpectedly before finishing this task. "
-                "Check python-core-worker-*.log files for more information.")
+        return ("The actor died unexpectedly before finishing this task."
```

Comment:
Please change to 
```
The actor died unexpectedly before finishing this task: {}
```

Reply:
I tried this out and it looks kind of ugly:
```
>>> ray.get(a.f.remote())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/eric/Desktop/ray/python/ray/_private/client_mode_hook.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/eric/Desktop/ray/python/ray/worker.py", line 1450, in get
    raise value
ray.exceptions.RayActorError: The actor died unexpectedly before finishing this task. Dead info: cancelling task of dead actor, dead_info=Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 587, in ray._raylet.task_execution_handler
  File "python/ray/_raylet.pyx", line 445, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 554, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 482, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 485, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 495, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 443, in ray._raylet.execute_task.function_executor
  File "/home/eric/Desktop/ray/python/ray/function_manager.py", line 556, in actor_method_executor
    return method(__ray_actor, *args, **kwargs)
  File "<stdin>", line 4, in __init__
AssertionError
An unexpected internal error occurred while the worker was executing a task.
```

Can you fix this to print the same thing as it does currently? Which is:
```
>>> ray.get(a.f.remote())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ray/anaconda3/lib/python3.6/site-packages/ray/_private/client_mode_hook.py", line 47, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 1408, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(AssertionError): ray::A.f() (pid=2410, ip=172.31.20.194)
  File "python/ray/_raylet.pyx", line 450, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 483, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 486, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 496, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 443, in ray._raylet.execute_task.function_executor
  File "<stdin>", line 4, in __init__
```

Reply:
> I tried this out and it looks kind of ugly:
> Can you fix this to print the same thing as it does currently? Which is:

To clarify, since the exception is stored as string now(because cross-language exception serialization is difficult). I can't set it as a cause of the RayActorError.

The only thing I can do is to add a new line before the exception string, can you accept this?

Reply:
Or I can store both the serialized exception and the raw string to GCS.
When the caller and callee are in the same language, the exception will be propagated correctly.
When the caller and callee are in different language, the exception will be shown as raw string.